### DISCUSSION TITLE
Same request name but different method

### DISCUSSION THREAD

**Discussion by thisjaymaloney** [usebruno/bruno#1858] [General]
I am trying to use the same request name with different methods. I have a simple CRUD api that I would to name like this:

GET employee
PUT employee
DELETE employee

It seems like Bruno does not allow this. If you try to clone the first one, you don't get the option to change the method. If you try to create a new request (same name, different method), you get an error - 'duplicate request name not allowed'.

Thanks.

**Comment by Mertzenich** [comment]
You seem to be able to work around this by creating the request using a new name and then manually editing the bru file. The requests will then be displayed in the interface with the same name while the underlying files will be different.

![bruno-same-name](https://github.com/usebruno/bruno/assets/62816361/52d1f1b7-21dd-4b46-bf5f-0b18a54ff257)

**Reply by wormhoudt** [reply]
Thanks for the workaround.

**Comment by afrianjunior** [comment]
Can you explain why this feature more important?

I think you can do this to tackle the limitation:
* Use prefix, ex: 
  * Get Employee
  * Put Employee
* or more intuitive
  * Get List Employee
  * Get Detail Employee
  * Put One Employee
  * Put Many Employee
* if you make it intuitive, you didn't need prefix anymore
  * [GET] List Employee
  * [GET] Detail Employee
  * [PUT] One Employee
  * [PUT] Many Employee

**Comment by badlydrawnrob** [comment]
Yeah this seems an oversight to me, you call the urls by the same name so why not the request name? I'd add to that with a suggestion that generated `File Names.bru` should always be `snake-case.bru`, regardless of the request names. Happy to see Bruno as an option as Postman feels too bloated and confusing to learn.

**Comment by zing-rsa** [comment]
Bumping this. I think it's important to be able to have duplicate names since there will be different HTTP verbs for the same url. I don't want to have to prefix the name with `[GET]` or something because the UI already displays the `GET` in front of the name of the request. Perhaps bruno could allow a duplicate name through config(in yaml), and then smartly maintain the name of the underlying file so it isn't a duplicate in the filesystem. Basically the first 'workaround' suggestion but managed by bruno.

**Reply by maarten-gorilla-co** [reply]
fully agree. this should be automatically managed by bruno in my opinion. starting to manage the filenames and request names is for me a reason not to adopt bruno. REST was always meant to be self-describing by relying on HTTP methods and resource names, so from that perspective it would make sense the filenames carry that same naming convention.
